symbolic-il2cpp/src/usym.rs

```diff
@@ -477,12 +471,14 @@ impl<'a> UsymSymbols<'a> {
     pub fn lookup_source_record(&self, ip: u64) -> Option<UsymSourceRecord> {
         // TODO: need to subtract the image base to get relative address
         match self.records.binary_search_by_key(&ip, |r| r.address) {
-            Ok(index) => self.resolve_record(index),
-            Err(index) => self.resolve_record(index - 1),
+            Ok(index) => self.get_record(index),
+            Err(index) => self.get_record(index - 1),
         }
     }
 
-    // TODO: Add iterator over records?
+    pub fn records(&'a self) -> impl Iterator<Item = UsymSourceRecord<'a>> {
```

Comment:
Some people say you should never return an un-typed iterator (i.e. one that's not a public struct) in a public api.  I think I agree with those people, though I don't mind here as all of this is experimental code anyway so please leave it as is for now.

Reply:
will do 👍 i'll briefly play around with the types to see what it would look like if i did create a dedicated type for this, but it'll just be in a separate branch/PR. thanks for calling this out!

Reply:
In this case, the dedicated type would just be a newtype around `std::FilterMap<std::slice::Iter<…>,…>`, but then again the `FilterMap` has an unnamable closure :-(

If we wait some more Rust versions, we will eventually have TAIT (type alias impl trait), aka a type alias (named typed) to an `impl trait` which itself is unname-able ;-)

Reply:
Though a newtype is still preferable over type alias is it not?  It means you have control over the type and can more easily evolve it without braking the API.

Reply:
TAIT offers the same thing. Its a nameable, but opaque type. So you do have control over it, and can evolve it over time. Either way, its nightly only for now so we won’t be able to use it in the foreseeable future anyway ;-)

Reply:
i've left a comment to revisit this at some point, but i guess no real promises about what'll happen here unless we decide to go with this solution. then we'll definitely have to clean this up.